spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/StreamListenerAnnotationBeanPostProcessor.java

```diff
@@ -208,25 +208,25 @@ private boolean checkDeclarativeMethod(Method method, String methodAnnotatedInbo
 						.getValue(methodParameter.getParameterAnnotation(Output.class));
 				Assert.isTrue(StringUtils.hasText(outboundName), StreamListenerErrorMessages.INVALID_OUTBOUND_NAME);
 				Assert.isTrue(
-						isDeclarativeMethodParameter(this.applicationContext.getBean(outboundName), methodParameter),
+						isDeclarativeMethodParameter(this.applicationContext.getBean(outboundName), methodParameter, true),
 						StreamListenerErrorMessages.INVALID_DECLARATIVE_METHOD_PARAMETERS);
 				return true;
 			}
 			if (StringUtils.hasText(methodAnnotatedOutboundName)) {
 				return isDeclarativeMethodParameter(this.applicationContext.getBean(methodAnnotatedOutboundName),
-						methodParameter);
+						methodParameter, false);
 			}
 			if (StringUtils.hasText(methodAnnotatedInboundName)) {
 				return isDeclarativeMethodParameter(this.applicationContext.getBean(methodAnnotatedInboundName),
-						methodParameter);
+						methodParameter, false);
 			}
 		}
 		return false;
 	}
 
-	private boolean isDeclarativeMethodParameter(Object targetBean, MethodParameter methodParameter) {
+	private boolean isDeclarativeMethodParameter(Object targetBean, MethodParameter methodParameter, boolean hasParameterAnnoation) {
 		if (targetBean != null) {
-			if (methodParameter.getParameterType().isAssignableFrom(targetBean.getClass())) {
+			if (hasParameterAnnoation && methodParameter.getParameterType().isAssignableFrom(targetBean.getClass())) {
```

Comment:
I don't think this will work properly. Consider this use case:

```
public interface KStreamProcessor {
	@Input("input")
	KStream<?, ?> input();

	@Output("output")
	KStream<?, ?> output();
}
```
and
```
@StreamListener("input")
@SendTo("output")
public KStream<?, WordCount> process(KStream<?, String> input) {
// omitted
}
```

This change will prevent the parameter from being treated as declarative, won't it?